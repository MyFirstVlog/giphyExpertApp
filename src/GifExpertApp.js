import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const categoriasRandom = ['Parasite', 'One Punch', 'Tokyo Ghoul']
let init = ['Naruto', 'Attack On Titans', 'Kimetzu no Yaiba']

const GifExpertApp = ({defaultCategories = []}) => {


    const [categorias, setCategorias] = useState(defaultCategories)


    // const handleAdd = () => {
    //     const categoriasRandomNro = Math.floor(Math.random()*3)
    //     const valorAPoner = categoriasRandom[categoriasRandomNro]
    //     setCategorias([valorAPoner,...categorias])
    // }

    return(
        <>
            <h2>GifExpertApp</h2>

            <AddCategory setCategorias={setCategorias}/>
            <hr/>

            <ol>
                {
                    categorias.map((each) =>
                    <GifGrid 
                        key={each}
                        category={each}
                    />)
                }
            </ol>
        </>
    )
} 

export default GifExpertApp