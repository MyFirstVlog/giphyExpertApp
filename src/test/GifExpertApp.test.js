import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas generales a la app gifExpertApp', () => {
    
    test('debe hacer match con el wrapper el snapshot ', () => {
        const wrapper = shallow(<GifExpertApp  />)
        
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostart una lista de categorias ', () => {
        const categorias = ['CR7', 'Real Madrid']
        const wrapper = shallow(<GifExpertApp defaultCategories={categorias} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categorias.length)
    })
    
    
})
