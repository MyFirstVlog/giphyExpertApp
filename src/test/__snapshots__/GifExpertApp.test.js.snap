// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas generales a la app gifExpertApp debe hacer match con el wrapper el snapshot  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas generales a la app gifExpertApp debe mostart una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="CR7"
      key="CR7"
    />
    <GifGrid
      category="Real Madrid"
      key="Real Madrid"
    />
  </ol>
</Fragment>
`;
